packages/messaging/README.md

```diff
@@ -127,10 +136,246 @@ Handling push notifications must be done in native (Android/iOS) code for the Re
+
+
+**Syntax**
+```javascript
+clearMessage();
+```
+
+**Example**
+```javascript
+var message: Message;
+message.clearMessage();
+```
+
+## Programatically control the display of in-app messages
+App developers can now create a type MessagingDelegate in order to be alerted when specific events occur during the lifecycle of an in-app message.
+
+Definition of type MessagingDelegate is:
+```javascript
+type MessagingDelegate = {    
+
+    onShow(message: Message): void;
+    
+    onDismiss(message: Message): void;
+   
+    shouldShowMessage(message: Message): boolean;
+
+    urlLoaded(url: string, message: Message): void;
+};
+```
+Object of type MessagingDelegate can be create as shown below:
+```javascript
+const messagingDelegate = {
+
+    onShow(message: Message) {
+       // Action after message is displayed.
+    },
+
+    onDismiss(message: Message) {
+       // Action after message is dismissed.
+    },
+
+    shouldShowMessage(message: Message) {                  
+        return true; //Return true to show the Message else return false
+    },
+
+    urlLoaded(url: string, message: Message) {
+       // Action after message loads an URL
+    }
+  };
+```
+
+### Controlling when the message should be shown to the end user.
+If a MessagingDelegate has been set, the delegate's shouldShowMessage method will be called prior to displaying an in-app message for which the end user has qualified. The developer is responsible for returning true if the message should be shown, or false if the message should be suppressed.
+
+Below is an example of when the developer may choose to suppress an in-app message due to the status of some other workflow within the app:
+```javascript
+function shouldShowMessage(message: Message): boolean {
+    if(someOtherWorkflowStatus == "inProgress") {
+        return false
+    }
+
+    return true
+}
+```
+
+Another option for the developer is to store a reference to the Message object, and call the show() method on it at a later time. App developer also have to call _Messaging.saveMessage(message)_ from _shouldShowMessage_ for caching the Message on the native side of the RN package.
+Continuing with the above example, the developer has stored the message that was triggered initially, and chooses to show it upon completion of the other workflow:
+
+```javascript
+var currentMessage: Message;
+
+function otherWorkflowFinished() {
+    anotherWorkflowStatus = "complete";
+    currentMessage.show();
+}
+
+function shouldShowMessage(message: Message): boolean {
+    if(someOtherWorkflowStatus == "inProgress") {                
+        // store the current message for later use
+        Messaging.saveMessage(message);
+        currentMessage = message;
+        return false
+    }
+
+    return true
+}
```

Comment:
i think we're using different names for the same variable in the example: 

`anotherWorkflowStatus` and `someOtherWorkflowStatus`